semantic_segmentation/code/pretrain.py

```diff
@@ -1,21 +1,14 @@
-from comet_ml import Experiment
 import torch
-import torch.nn as nn
 import torchvision
-from torchvision import transforms
 from torch.utils.data import DataLoader
-import torch.nn.functional as F
 import models
-import argparse
+from args import get_args
+from dataset import CamVid, Cityscapes
 from helper import *
-torch.cuda.set_device(0)
+from trainer import pretrain
 
-parser = argparse.ArgumentParser(description = 'Standalone training of UNet based on ResNet encoder')
-parser.add_argument('-m', choices = ['resnet10', 'resnet14', 'resnet18', 'resnet20', 'resnet26', 'resnet34'], help = 'Give the encoder name from the choices')
-parser.add_argument('-d', choices = ['camvid', 'cityscapes'], help = 'Give the dataset to be used for training from the choices')
-parser.add_argument('-e', type = int, help = 'Give number of epochs for training')
-parser.add_argument('-s', type = int, help = 'Give the random seed number')
-args = parser.parse_args()
+torch.cuda.set_device(0)
```

Comment:
we can add an argument for GPU also. But, don't know how useful it will be.

Reply:
Even though there is no point in using cpu, it's better to include `device = torch.device("cuda" if torch.cuda.is_available() else "cpu")`. Helps during debugging.

Reply:
yes, do `"cuda:0"` to be more specific about first GPU. But then argument might be useful.

Reply:
i have added argument for that